assembl/static2/js/app/components/search.jsx

```diff
@@ -252,12 +275,27 @@ const UserHit = (props) => {
   );
 };
 
-const IdeaHit = (props) => {
+const DumbIdeaHit = (props) => {
+  const locale = props.locale;
   const source = props.result._source;
-  const shortTitle = highlightedTextOrTruncatedText(props.result, 'short_title');
-  const definition = highlightedTextOrTruncatedText(props.result, 'definition');
-  const announceTitle = highlightedTextOrTruncatedText(props.result, 'title');
-  const announceBody = highlightedTextOrTruncatedText(props.result, 'body');
+  let shortTitle = getFieldAnyLang(props.result.highlight, 'title', locale);
+  if (!shortTitle) {
+    shortTitle = getFieldAnyLang(source, 'title', locale);
+  }
+  let definition = getFieldAnyLang(props.result.highlight, 'description', locale);
+  if (!definition) {
+    definition = getFieldAnyLang(source, 'description', locale);
+  }
+  // long title missing?
+  let announceTitle = getFieldAnyLang(props.result.highlight, 'announcement_title', locale);
+  if (!announceTitle) {
+    announceTitle = getFieldAnyLang(source, 'announcement_title', locale);
+  }
+  let announceBody = getFieldAnyLang(props.result.highlight, 'announcement_body', locale);
+  if (!announceBody) {
+    announceBody = getFieldAnyLang(source, 'announcement_body', locale);
+  }
+
```

Comment:
Not exactly the same as before. Before, when the text came from source, not source.highlight, I truncated it because it could be too long.

Reply:
So in _other you can have several languages? So if you indexed a post japanese and chinese, we will have those two languages in body_other and we show a post in japanese and chinese in the result?

Reply:
Ok, so not really an issue, we just ad de, zh, ja in the indexed locales

Reply:
As discussed: I think it's an improvement over the previous situation, where und was lost. It is robust against mis-identified langstrings. The workaround is to add languages in the index.